### DISCUSSION TITLE
File excluded from the override?

### DISCUSSION THREAD

**Discussion by iorbita** [templaza/astroid-framework#547] [Q&A]
Hi,
I thought that files in **/templates/astroid_template_one/html/com_users** were safe from any changes since they are located in the html folder of the template. 
Since _astroid_template_one_ template has been updated, it seems that **/templates/astroid_template_one/html/com_users/login/default.php** file has also been updated and I've lost all the changes I made in this file (fortunately I had a backup).

Should I assume that this file will always be modified every time the template will be updated?

I thought that files in template's html folder were safe from modification, is this an exception?

Thanks,
Lorenzo

**Comment by sonvnn** [comment]
You can use child-template to save your override.

**Comment by iorbita** [comment]
Hi Sonny,
so to be more precise, this would mean that I can copy the whole html folder (and subfolders) into **/media/templates/site/astroid_template_one** ?


![astroid-override](https://github.com/templaza/astroid-framework/assets/566221/246e6f4b-8a50-4987-aebd-bec5f4454c7c)

**Comment by Chacapamac** [comment]
> I can copy the whole html folder (and subfolders) into /media/templates/site/astroid_template_one ?
Never try that, also curious if this will work? — My guess is NO as the system probably look for the html folder in the template folder and not in the media folder.

@iorbita , if you know what template > html modifications do and you have a test website, try it, move your **html** folder to the **media** folder and check if your modifications still take place.


> I thought that files in template's html folder were safe from modification, is this an exception?
Yeah, me to, I was thinking that html/overrides wasn’t updated. I guess the one that are pre-made by Astroid need to be updated from time to time, even us we need to update them as Joomla/Astroid changes.

Yeah, Child Template will be the most controllable ultimate override.

**• Basics Notion on Child Templates**_____________________________________________________________________________
If you are familiar with the JoomlaRoot > templates > Your_template > HTML folder override system, you know that you can customize many Joomla core layout files and keep the modifications in this folder even if Joomla is updated.

Now, with the “Child Template” ability, you can do the same type of override with ALL files making up a template. It permits you to change anything in a template and keep these changes even if you update the Parent Template. 

When creating a “Child Template” another template folder is created with basically nothing else than a templateDetails.xml file. At this point, the “Child Template” take All of the Parent files and settings to construct its layout.

Any settings changes in the “Child Templates” or added “Parent” corresponding folder/files will override the “Parent”. This way you can override only the settings of the “Child” (Template Manager) or start to override ANYTHING from the “Parent” in the “Child Template” folder. When you update the “Parent” you are also updating the “Child template”, but all your “Child” settings and folder/files override stays untouched.

Now, you have the power to completely modified a Template or simply changed some settings without worrying braking or touching the Parent Template.

**- If you want to dive deeper into Joomla Child Templates**
https://magazine.joomla.org/all-issues/may-2022/a-deep-dive-into-joomla-child-templates

**- If you want a fast video overview how to create one**
https://www.youtube.com/watch?v=_3cUTEUvczk

**Comment by iorbita** [comment]
ok thanks for the reply, so it looks like if I add an html folder in the child template the files inside won't be updated with the next template update... I'll try it, thanks again.

**Reply by Chacapamac** [reply]
Absolutely, a Child template is the best way to do and control any customization to a template. This is the PRO way.

**Comment by rinka88** [comment]
If you want the file to be saved, then give it a unique name, not a default
